package stream_switch;

import "pb_metadata.proto";

message ProtoMediaStatisticReq{
    //no param now
}


message ProtoSubStreamMediaStatistic{

    optional int32 sub_stream_index = 1; 
    optional ProtoSubStreamMediaType media_type = 2; //the type of this sub stream

    
    //about bytes
    optional uint64 total_bytes = 20;  //the bytes received of this sub stream
    optional uint64 key_bytes = 21; //the bytes in the key frames of this sub stream. key_frame_bytes <= received_bytes
    
    //about frame
	optional uint64 expected_frames = 30; // the frame number should be received 
    optional uint64 total_frames = 31;    // the received frame number
    optional uint64 key_frames = 32;      // the received key frame number
    optional uint64 last_gov = 33;           //last gov
} 


message ProtoMediaStatisticRep{
    optional uint32 ssrc = 1;
    optional int64 timestamp = 2;   //the statistic generation time
    optional uint64 sum_bytes = 3;  //the sum bytes received of all sub streams
    //tag below 64 is reserved to future extension
    
    repeated ProtoSubStreamMediaStatistic sub_stream_stats = 64;   


}   


